spec/user_spec.rb

```diff
@@ -0,0 +1,12 @@
+require_relative '../lib/user'
+
+RSpec.describe User do
+  let(:input) {User.new}
+
+  describe "#initialize" do
+    it "creates an instance of User" do
+      expect (input).to eq(true)
+    end
+  end
+
+end
```

Comment:
I'd recommend letting your tests drive your development. Your `user.rb` is doing some pretty heavy-duty work without tests. If you refactor it can make your job a lot harder without tests in place. I'd go back and test your public methods. If the methods/behavior is difficult to test, this might be a sign that you might need to re-write your code to be more testable. In POODR, Metz talks about how tests should reflect the complexity of the code they are trying to test. Strive to have easy to understand tests. You'll find that it makes your code easier to reason about, refactor and explain.